gitlab_integration/README.md

```diff
@@ -1,16 +1,34 @@
 # Gitlab integration research
 
+## Project integrations
+
+It allow you to integrate GitLab with other applications. They are a bit like plugins in that they allow a lot of freedom in adding functionality to GitLab.
+
+Just like Github Apps but with some limited functionality. This will be a very easy way for any projects to use the Packit service (eg in https://gitlab.gnome.org)
+
+So just by clicking one button, packit service will be active for the project - 
+
+![packit service example](./gitlab-packit.png)
+
+The above method is much easier for the user rather then creating a webhook manually.
+
+![Webhook](./webhook.png)
+
 ## Pros and Cons of having Packit as a Project Service in Gitlab
 
-A integration / project service in gitlab works in a way that the service does not have the access to the repository and can only work as a webhook to report changes based on permissions given (issues, merge requests etc) 
+A integration / project service in gitlab works in a way that the service does not have the access to the repository and can only work as a webhook to report changes based on permissions given (issues, merge requests etc).
 
 ### Pros 
 
 - Users dont have to setup webhook for each project.
-- Instead of tweaking the config file for packit you can have a settings page for the service.
+- In the later stages we can setup a settings page for the packit-service in gitlab which will act as a config for the service instead of tweeking packit config file in the repository.
 
 ### Cons
 
-- Users still have to define auth for hooks (Packit).
-- Users will still have to give access to a packit service user for getting the build updates.
+- Users will still have to give access to a packit service "user" for getting the build updates.
+ - This introduces a new programming language "Ruby" that we will need to work on for maintaining the service in gitlab.
+
+For adding the service to gitlab we will have to contribute to the https://gitlab.com/gitlab-org/gitlab which has to be approved by the admins (Currently looking more into it). 
```

Comment:
I would make a specific section for the implementation. (So we can clearly say if it is worth spending the time on it.)

Reply:
okay!